src/model_signing/_signing/signing.py

```diff
@@ -120,6 +126,46 @@ def dsse_payload_to_manifest(dsse_payload: dict[str, Any]) -> manifest.Manifest:
     return manifest.Manifest(model_name, items, serialization)
 
 
+def dsse_payload_to_manifest_compat(
+    dsse_payload: dict[str, Any],
+) -> manifest.Manifest:
+    """Builds a manifest from the DSSE payload read from a signature.
+
+    This is the same as `dsse_payload_to_manifest` but using a DSSE payload as
+    defined at v0.2 release. This was experimental but got used in production
+    before v1.0 so we need to patch support to it while verifiers migrate to the
+    forward compatible format defined by v1.0.
+
+    Args:
+        payload: The in-toto DSSE envelope to convert to manifest.
+
+    Returns:
+        A manifest representing the signed model.
+
+    Raises:
+        ValueError: The payload cannot be deserialized to a manifest.
+    """
+    # Model name is not defined, use a constant.
+    model_name = "compat-undefined-not-present"
+
+    # Serialization format is not present, build a fake one.
+    serialization = manifest.SerializationType.from_args(
+        {"method": "files", "hash_type": "sha256", "allow_symlinks": "false"}
```

Comment:
these fakes match the defaults from the old signatures? Though `__eq__` doesnt care about that, so I guess it doesn't matter.

Reply:
Yes, they should match the original from the signature. Without these fakes, we'd need to call the "private" constructor of the serialization type, but this is cleaner.